examples/99-advanced/planets.py

```diff
@@ -0,0 +1,119 @@
+
+"""
+import pyvista
+from pyvista import examples
+
+###############################################################################
+# Celestial bodies with the Milky Way in the background.
+
+
+# Light of the sun in the +x axis direction.
+
+light = pyvista.Light()
+light.set_direction_angle(30, -20)
+
+# Load planet
+
+# https://tamaskis.github.io/files/Visualizing_Celestial_Bodies_in_3D.pdf
+# Mercury's radius is 2439.0 km
+mercury = examples.planets.load_mercury(radius=2439.0, lat_resolution=150, lon_resolution=300)
+# Venus's radius is 6052.0 km
+venus = examples.planets.load_venus(radius=6052.0, lat_resolution=150, lon_resolution=300)
+# Mars's radius is 3397.2 km
+mars = examples.planets.load_mars(radius=3397.2, lat_resolution=150, lon_resolution=300)
+# Jupiter's radius is 71492.0 km
+jupiter = examples.planets.load_jupiter(radius=71492.0, lat_resolution=150, lon_resolution=300)
+# Saturn's radius is 60268.0 km
+saturn = examples.planets.load_saturn(radius=60268.0, lat_resolution=150, lon_resolution=300)
+# Saturn's rings range from 7000.0 km to 80000.0 km from the surface of the planet
+inner = 60268.0 + 7000.0
+outer = 60268.0 + 80000.0
+saturn_ring_alpha = examples.planets.load_saturn_ring_alpha(inner=inner, outer=outer, c_res=50)
+# Uranus's radius is 25559.0 km
+uranus = examples.planets.load_uranus(radius=25559.0, lat_resolution=150, lon_resolution=300)
+# Neptune's radius is 24764.0 km
+neptune = examples.planets.load_neptune(radius=24764.0, lat_resolution=150, lon_resolution=300)
+
+# Move planet position (Numbers have no meaning. The planets are laid out for
+# easy viewing.).
+
+mercury.translate((0.0, 0.0, 0.0), inplace=True)
+venus.translate((-15000.0, 0.0, 0.0), inplace=True)
+mars.translate((-45000.0, 0.0, 0.0), inplace=True)
+jupiter.translate((-150000.0, 0.0, 0.0), inplace=True)
+saturn.translate((-400000.0, 0.0, 0.0), inplace=True)
+saturn_ring_alpha.translate((-400000.0, 0.0, 0.0), inplace=True)
+uranus.translate((-600000.0, 0.0, 0.0), inplace=True)
+neptune.translate((-700000.0, 0.0, 0.0), inplace=True)
+
+# Add planets to Plotter.
+
+plotter = pyvista.Plotter(lighting="none")
+cubemap = examples.download_cubemap_space_16k()
+_ = plotter.add_actor(cubemap.to_skybox())
+plotter.set_environment_texture(cubemap, True)
+plotter.add_light(light)
+plotter.add_mesh(mercury)
+plotter.add_mesh(venus)
+plotter.add_mesh(mars)
+plotter.add_mesh(jupiter)
+plotter.add_mesh(saturn)
+plotter.add_mesh(saturn_ring_alpha)
+plotter.add_mesh(uranus)
+plotter.add_mesh(neptune)
+plotter.camera.zoom(1.5)
+plotter.show()
+
+###############################################################################
+# `Textures in this pack are based on NASA <https://www.solarsystemscope.com/textures/https://www.solarsystemscope.com/textures/>`_
+# elevation and imagery data. Colors and shades of the textures are tuned
+# according to true-color photos made by Messenger, Viking and Cassini
+# spacecrafts, and, of course, the Hubble Space Telescope.
```

Comment:
This also seems to be copy-pasted from https://www.solarsystemscope.com/textures/. I'd redact and rephrase this, just point out that the textures are from there and it's based on NASA data.

Reply:
See eca4350c66e97593b847e4650c608df2d7a5019e .